src/internal/database/tenant.ts

```diff
@@ -205,13 +206,22 @@ export async function getServiceKey(tenantId: string): Promise<string> {
  */
 export async function getJwtSecret(
   tenantId: string
-): Promise<{ secret: string; jwks: TenantConfig['jwks'] | null }> {
```

Comment:
The `tenant.ts` file is getting a bit out hands 😄 
We might need to consider to start refactoring this. We can do it either in this PR (at least for the new part) or after we merge It.

The part to refactor are: 

- S3 keys manager 
- JWKS generator manager


We probably need to use a simple pattern with 2 classes:

```
interface JWKSManagerStore {
  create()
  delete()
  ....etc
}
class JWKSManager {
  constructor(storage: JWKSManagerStore) {}
}

class JWKSManagerStoreKnex {
   constructor(knex: Knex)
}
```
 
a similar pattern for S3Credentials:

```ts
interface S3CredentialStore {
   create()
   delete()
   ...etc
}

class S3CredentialManager {
   constructor(store: S3CredentialStore) 
}
```

Reply:
Sounds good. I was planning to fix that little cache bug I found in the S3 Credentials and add test coverage for it as there is none currently. I think it makes sense to do this refactor as part of that PR.